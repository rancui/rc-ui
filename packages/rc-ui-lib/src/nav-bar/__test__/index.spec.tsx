import React from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react';
import NavBar from '..';
import Icon from '../../icon';
import { mockGetBoundingClientRect } from '../../utils/dom/mock';

describe('NavBar', () => {
  afterEach(() => {
    cleanup();
    jest.restoreAllMocks();
  });

  it('should render left area correct', async () => {
    const { container } = render(<NavBar leftArea="speaker-s" title="foo" />);
    expect(container).toMatchSnapshot();
  });

  it('should render right area correct', async () => {
    const { container } = render(<NavBar rightArea="speaker-s" title="foo" />);
    expect(container).toMatchSnapshot();
  });

  it('should render ReactNode icon correct', async () => {
    const { container } = render(
      <NavBar leftArea={<Icon name="cart-o" color="#1989fa" />} title="foo" />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render leftArrow correct', async () => {
    const { container } = render(<NavBar leftArrow title="foo" />);
    expect(container).toMatchSnapshot();
  });

  it('should render correct when set props fixed and placeholder', async () => {
    const restore = mockGetBoundingClientRect({ height: 50 });
    const { container } = render(<NavBar fixed placeholder title="foo" />);
    expect(container).toMatchSnapshot();
    restore();
  });

  it('should emit click-left event when clicking left text', async () => {
    const onClickLeft = jest.fn();
    const { container } = render(<NavBar title="foo" leftArea="left" onClickLeft={onClickLeft} />);
    const item = container.querySelector('.rc-nav-bar__left');
    await fireEvent.click(item);
    expect(onClickLeft).toHaveBeenCalledTimes(1);
  });

  it('should emit click-right event when clicking right text', async () => {
    const onClickRight = jest.fn();
    const { container } = render(
      <NavBar title="foo" rightArea="right" onClickRight={onClickRight} />,
    );
    const item = container.querySelector('.rc-nav-bar__right');
    await fireEvent.click(item);
    expect(onClickRight).toHaveBeenCalledTimes(1);
  });

  it('should have safe-area-inset-top class when using safe-area-inset-top prop ', async () => {
    const { container } = render(<NavBar title="foo" safeAreaInsetTop />);

    expect(container).toMatchSnapshot();
  });

  it('should change z-index when using zIndex prop', async () => {
    const { container } = render(<NavBar title="foo" zIndex={100} />);

    expect(getComputedStyle(container.querySelector('.rc-nav-bar')).zIndex).toEqual('100');
  });
});
