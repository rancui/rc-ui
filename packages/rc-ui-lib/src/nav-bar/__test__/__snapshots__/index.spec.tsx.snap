// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should have safe-area-inset-top class when using safe-area-inset-top prop  1`] = `
<div>
  <div
    class="rc-nav-bar rc-nav-bar--safe-area-inset-top rc-hairline--bottom"
    style="z-index: 1;"
  >
    <div
      class="rc-nav-bar__content"
    >
      
      <div
        class="rc-nav-bar__title van-ellipsis"
      >
        foo
      </div>
      
    </div>
  </div>
</div>
`;

exports[`NavBar should render ReactNode icon correct 1`] = `
<div>
  <div
    class="rc-nav-bar rc-hairline--bottom"
    style="z-index: 1;"
  >
    <div
      class="rc-nav-bar__content"
    >
      <div
        class="rc-nav-bar__left rc-haptics-feedback"
      >
        <span
          class="rc-nav-bar__text"
        >
          <i
            class="rc-badge__wrapper van-icon van-icon-cart-o"
            style="color: rgb(25, 137, 250);"
          />
        </span>
      </div>
      <div
        class="rc-nav-bar__title van-ellipsis"
      >
        foo
      </div>
      
    </div>
  </div>
</div>
`;

exports[`NavBar should render correct when set props fixed and placeholder 1`] = `
<div>
  <div
    class="rc-nav-bar__placeholder"
    style="height: 50px;"
  >
    <div
      class="rc-nav-bar rc-nav-bar--fixed rc-hairline--bottom"
      style="z-index: 1;"
    >
      <div
        class="rc-nav-bar__content"
      >
        
        <div
          class="rc-nav-bar__title van-ellipsis"
        >
          foo
        </div>
        
      </div>
    </div>
  </div>
</div>
`;

exports[`NavBar should render left area correct 1`] = `
<div>
  <div
    class="rc-nav-bar rc-hairline--bottom"
    style="z-index: 1;"
  >
    <div
      class="rc-nav-bar__content"
    >
      <div
        class="rc-nav-bar__left rc-haptics-feedback"
      >
        <span
          class="rc-nav-bar__text"
        >
          speaker-s
        </span>
      </div>
      <div
        class="rc-nav-bar__title van-ellipsis"
      >
        foo
      </div>
      
    </div>
  </div>
</div>
`;

exports[`NavBar should render leftArrow correct 1`] = `
<div>
  <div
    class="rc-nav-bar rc-hairline--bottom"
    style="z-index: 1;"
  >
    <div
      class="rc-nav-bar__content"
    >
      <div
        class="rc-nav-bar__left rc-haptics-feedback"
      >
        <i
          class="rc-badge__wrapper rc-nav-bar__arrow van-icon van-icon-arrow-left"
        />
        
      </div>
      <div
        class="rc-nav-bar__title van-ellipsis"
      >
        foo
      </div>
      
    </div>
  </div>
</div>
`;

exports[`NavBar should render right area correct 1`] = `
<div>
  <div
    class="rc-nav-bar rc-hairline--bottom"
    style="z-index: 1;"
  >
    <div
      class="rc-nav-bar__content"
    >
      
      <div
        class="rc-nav-bar__title van-ellipsis"
      >
        foo
      </div>
      <div
        class="rc-nav-bar__right rc-haptics-feedback"
      >
        <span
          class="rc-nav-bar__text"
        >
          speaker-s
        </span>
      </div>
    </div>
  </div>
</div>
`;
